src/schema/v2/users/updateUserSaleProfileMutation.ts

```diff
@@ -0,0 +1,79 @@
+import { GraphQLNonNull, GraphQLString } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import { snakeCase } from "lodash"
+
+interface Input {
+  id: string
+  addressLine1?: string
+  addressLine2?: string
+  city?: string
+  state?: string
+  zip?: string
+  country?: string
+}
+
+interface GravityInput {
+  id: string
+  address_1?: string
+  address_2?: string
+  city?: string
+  state?: string
+  zip?: string
+  country?: string
+}
+
+interface GravityError {
+  statusCode: number
+  body: { error?: string; text?: string; message?: string }
+}
+
+export const updateUserSaleProfileMutation = mutationWithClientMutationId<
+  Input,
+  any | null,
+  ResolverContext
+>({
+  name: "UpdateUserSaleProfileMutation",
+  description: "Update the user sale profile",
+  inputFields: {
+    id: { type: new GraphQLNonNull(GraphQLString) },
+    addressLine1: { type: GraphQLString },
+    addressLine2: { type: GraphQLString },
+    city: { type: GraphQLString },
+    state: { type: GraphQLString },
+    zip: { type: GraphQLString },
+    country: { type: GraphQLString },
+  },
+  outputFields: {},
+  mutateAndGetPayload: async (args, { updateUserSaleProfileLoader }) => {
+    if (!updateUserSaleProfileLoader) {
+      throw new Error(
+        "You need to be signed in as an admin to perform this action"
+      )
+    }
+
+    const initialValue = {} as GravityInput
+    const gravityOptions = Object.keys(args)
+      .filter((key) => key !== "id")
+      .reduce((acc, key) => {
+        if (key === "addressLine1") {
+          return { ...acc, address_1: args[key] }
+        } else if (key === "addressLine2") {
+          return { ...acc, address_2: args[key] }
+        } else {
+          return { ...acc, [snakeCase(key)]: args[key] }
+        }
+      }, initialValue)
```

Comment:
`non-blocking` This looks rather complicated for what could just be manually written out

```tsx
const gravityOptions = {
  address_1: args.addressLine1,
  // ...etc
}
```

Reply:
+1. dumb, redundant and literal wins the game in graphql

Reply:
Dumb coming right up! [bda9fe5](https://github.com/artsy/metaphysics/pull/3952/commits/bda9fe5408afbf2ae24299f15f570e1e070dac1a)